SciPy v1.14.0 breaks POT's use of scipy.optimize.scalar_search_armijo

With the release of [`scipy` `v1.14.0`](https://pypi.org/project/scipy/1.14.0/) there's an issue with with POT's use of 

```python
from scipy.optimize import scalar_search_armijo
```

```pytb
==================================== ERRORS ====================================
______________________ ERROR collecting tests/test_emd.py ______________________
ImportError while importing test module '/home/runner/work/Wasserstein/Wasserstein/tests/test_emd.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/ot/optim.py:21: in <module>
    from scipy.optimize import scalar_search_armijo
E   ImportError: cannot import name 'scalar_search_armijo' from 'scipy.optimize' (/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/scipy/optimize/__init__.py)

During handling of the above exception, another exception occurred:
/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_emd.py:2: in <module>
    import ot
/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/ot/__init__.py:23: in <module>
    from . import optim
/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/ot/optim.py:23: in <module>
    from scipy.optimize.linesearch import scalar_search_armijo
E   ImportError: cannot import name 'scalar_search_armijo' from 'scipy.optimize.linesearch' (/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/scipy/optimize/linesearch.py)
```

This needs to get fixed upstream, so for the time being we can ignore `scipy` `v1.14.0`.

--- a/setup.cfg
+++ b/setup.cfg
@@ -77,5 +77,7 @@ install_requires =
 test =
     energyflow
     pot
+    # FIXME: https://github.com/thaler-lab/Wasserstein/issues/36
+    scipy!=1.14.0
     pytest
     coverage

